primitives/teerex/src/lib.rs

```diff
@@ -202,12 +202,26 @@ pub type Pcesvn = u16;
 pub type ShardIdentifier = H256;
 pub type SidechainBlockNumber = u64;
 
-#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, sp_core::RuntimeDebug, TypeInfo)]
-pub struct Request {
-	pub shard: ShardIdentifier,
-	pub cyphertext: Vec<u8>,
+// Litentry: use the name `RsaRequest` to differentiate from `AesRequest` (see aes_request.rs in
+// tee-worker) `Rsa` implies that the payload is RSA-encrypted (using enclave's shielding key)
+#[macro_export]
+macro_rules! decl_rsa_request {
```

Comment:
Maybe we can get rid of teerex's `call_worker` ?

Reply:
Hmm then there's no way to call arbitrary trusted call for II.

We didn't use `call_worker` because of https://github.com/litentry/tee-worker/issues/58

In general I think it still makes sense to keep it, maybe we want to rewrite it someday